synapse/handlers/auth.py

```diff
@@ -458,12 +458,9 @@ def _check_threepid(self, medium, authdict, password_servlet=False, **kwargs):
         identity_handler = self.hs.get_handlers().identity_handler
 
         logger.info("Getting validated threepid. threepidcreds: %r", (threepid_creds,))
-        if (
-            not password_servlet
-            or self.hs.config.email_password_reset_behaviour == "remote"
-        ):
+        if not password_servlet or self.hs.config.email_threepid_behaviour == "remote":
```

Comment:
Is doing this for non password reset still the correct thing to do?

Reply:
This will be changed in part 2 of https://github.com/matrix-org/synapse/pull/5835, where we hook up the `submit_token` stuffs.

Reply:
This is going to be changed in part 2 of https://github.com/matrix-org/synapse/pull/5835